lib/ansible/module_utils/common/parameters.py

```diff
@@ -600,7 +600,7 @@ def _validate_argument_types(argument_spec, parameters, prefix='', options_conte
             continue
 
         value = parameters[param]
-        if value is None:
+        if value is None and not spec.get('required') and spec.get('default') is None:
```

Comment:
2 other things to consider `choices` and type `raw`

Reply:
I have a test for raw validating that None is a valid value for a required raw option.

`choices` are checked after type coercion, and we don't validate `choices` match the `type`, so I think this improves internally consistency. This spec `dict(required=True, choices=[None, 'other_choice'])` would previously allow None. But it's ambiguous because the type is 'str', so imo it should be either
`dict(required=True, type='raw', choices=[None, 'other_choice'])`
or
`dict(required=True, choices=['None', 'other_choice'])`